src/vm/analysis/arithmetic_checker/tests.rs

```diff
@@ -133,7 +137,10 @@ fn test_variables_fail_arithmetic_check() {
 fn test_version_controlled_variables() {
     // Certain variables are controlled by the ClarityVersion. E.g., in Clarity1, 'tx-sponsor?'
     // is not a variable, so can be defined over.
-    let ok_tests = ["(define-private (foo) tx-sponsor?)"];
+    let ok_tests = [
+        "(define-private (foo) tx-sponsor?)",
+        "(define-private (foo) is-in-mainnet)",
```

Comment:
Same question here as with `tx-sponsor?` in your other PR -- why does this pass with `Clarity1` rules?

Reply:
It's because the arithmetic checker doesn't do type checking -- it's only looking for calls to native functions that are disallowed. In Clarity1, `is-in-mainnet` is a legal variable name that could be used as a function argument, for example. In Clarity2, however, it disqualifies a contract from being "arithmetic only". These example tests would fail the type checker, because they reference unbounded variables, but the arithmetic checker doesn't do variable dereferencing.

Reply:
Ah, gotcha. Thanks!